### DISCUSSION TITLE
Coorelation with logs generated via Lograge

### DISCUSSION THREAD

**Discussion by sharozem** [open-telemetry/opentelemetry-ruby#1289] [Q&A] [answered]
Hi, is there any way i can correlate the traces with rails logs generated via lograge gem using context propagation.

**Comment by plantfansam** [comment]
Hi! Thanks for raising this! We currently do not have the lograge gem instrumented (you can see all of our available instrumentation in the `instrumentation/` folder). If you'd like to see the lograge gem instrumented, definitely make an issue in this repository and include examples of the desired behavior 😄.

**Reply by sharozem** [reply]
ok.

**Comment by arielvalentin** [comment, marked as answer]
I have not used lograge before but I imagine you could do something like this:

```ruby
config.lograge.custom_options = lambda do |event|
  span = OpenTelemetry::Trace.current_span
  { "trace_id" => span.context.hex_trace_id, "span_id" => span.context.hex_span_id }
end
```

Or with the propagation API, however that will inject attributes for _all_ of the propagators you have configured, which may not be ideal: 

```ruby
config.lograge.custom_options = lambda do |event|
  {}.tap do |carrier| 
    OpenTelemetry.propagation.inject(carrier)
  end
end
```

**Reply by sharozem** [reply]
Thanks for the assist. Will test it and let u know if it works.

**Reply by sharozem** [reply]
I have tested the solution and now I am able to correlate logs and trace in grafana. thanks.
